source/panel_finder/panel_finder.py

```diff
@@ -17,12 +16,21 @@ def combined_panel(rect_a, rect_b):
 
 
 class PanelFinder(Module):
-    def __init__(self, state=None):
+    def __init__(self, state):
+        if state == "tensorflow":
```

Comment:
We should try to maintain continuity with the use of state and the data type of state. It gets pretty difficult to debug state issues if it can switch between a string and dictionary.

I think it would be better to just have an item in `state` determining what platform will be used to make inferences.
For example:
```
if state["platform"] == "tensorflow":
```

Reply:
I changed it to state["framework"] = "tensorflow"